src/sentry/models/file.py

```diff
@@ -16,27 +16,40 @@
 from jsonfield import JSONField
 from uuid import uuid4
 
-from sentry.db.models import BoundedPositiveIntegerField, Model
+from sentry.db.models import (
+    BoundedPositiveIntegerField, FlexibleForeignKey, Model
+)
+from sentry.utils.cache import Lock
 
 ONE_DAY = 60 * 60 * 24
 
 
-class File(Model):
+class FileBlob(Model):
     __core__ = False
 
-    name = models.CharField(max_length=128)
-    storage = models.CharField(max_length=128, null=True)
+    storage = models.CharField(max_length=128)
     storage_options = JSONField()
     path = models.TextField(null=True)
-    type = models.CharField(max_length=64)
     size = BoundedPositiveIntegerField(null=True)
-    checksum = models.CharField(max_length=40, null=True)
```

Comment:
I assume we'll need to backfill checksums for existing blobs if they don't have one, and will we want to reject null values moving forward after applying the unique index? I can't think of any reason that we wouldn't want a checksum since that is our form of deduping.